### PR TITLE
Add logic to screens ES-19

### PR DESCRIPTION
**Pull request by hassanSayed0**
register ( SetUserNameAndPassword ,  Verification , RegisterAccount) ViewModel storeAssemply
add three function to show ( SetUserNameAndPassword ,  Verification , RegisterAccount) it return viewControllerType with his dependences by use DependencyManager
add function to bind textField and viewModel in ( SetUserNameAndPassword ,  Verification , RegisterAccount) controller 
add function to update variable and check validate textField text  in  ( SetUserNameAndPassword ,  Verification , RegisterAccount) view model 
add function to validate text in validator to validate FullName ,PhoneNumber ,OneDigit and Email

### CODE REVIEW

File: e-store/e-store/ScenesScreen/Register/ProfilePassword/SetUserNameAndPasswordViewController.swift [hunk 1]
```diff
@@ -5,34 +5,89 @@
...
 }
 
 // MARK: Bind view model
 extension SetUserNameAndPasswordViewController {
-
+    private func bindViewModel() {
+        setUserNameAndPasswordViewModel.checkConfigButton { [weak self] enable in
+            self?.submitButton.isEnabled = enable
+            if enable {
+                self?.submitButton.selectStyle(style: .blueButton)
```
Comment (Mahmoud-Bassuni) [inline e-store/e-store/ScenesScreen/Register/ProfilePassword/SetUserNameAndPasswordViewController.swift:50]: its should handle from shared ui

File: e-store/e-store/ScenesScreen/Register/RegisterAccount/RegisterAccountViewModel.swift [hunk 2]
```diff
@@ -5,33 +5,70 @@
 //  Created by Hassan on 09/04/2023.
 //
 import Foundation
+import UIKit
```
Comment (Mahmoud-Bassuni) [inline e-store/e-store/ScenesScreen/Register/RegisterAccount/RegisterAccountViewModel.swift:8]: why import UIKit

File: e-store/e-store/ScenesScreen/Register/Verification/VerificationViewModel.swift [hunk 3]
```diff
@@ -4,29 +4,68 @@
 //
 //  Created by Hassan on 09/04/2023.
 //
-import Foundation
+import UIKit
```
Comment (Mahmoud-Bassuni) [inline e-store/e-store/ScenesScreen/Register/Verification/VerificationViewModel.swift:7]: the same

File: e-store/e-store/Utilities/Validator.swift [hunk 4]
```diff
@@ -7,18 +7,49 @@
...
     static func isValidUsername(username: String) -> Bool {
         //requires to contain lowercase and uppercase and numbers
         let usernameRegex = "^(?=.*[a-z])(?=.*[A-Z])(?=.*[0-9])[a-zA-Z0-9._-]{3,}$"
-            if let range = username.range(of: usernameRegex, options: .regularExpression) {
-                return username.distance(from: username.startIndex, to: range.lowerBound) == 0 &&
-                       username.distance(from: username.startIndex, to: range.upperBound) == username.count
-            }
-            return false
+        if let range = username.range(of: usernameRegex, options: .regularExpression) {
+            print(range)
```
Comment (Mahmoud-Bassuni) [inline e-store/e-store/Utilities/Validator.swift:17]: remove print

File: e-store/e-store/ScenesScreen/Register/ProfilePassword/SetUserNameAndPasswordViewController.swift (hunk 1 above)
Comment (hassanSayed0) [reply e-store/e-store/ScenesScreen/Register/ProfilePassword/SetUserNameAndPasswordViewController.swift]: done

File: e-store/e-store/ScenesScreen/Register/RegisterAccount/RegisterAccountViewModel.swift (hunk 2 above)
Comment (hassanSayed0) [reply e-store/e-store/ScenesScreen/Register/RegisterAccount/RegisterAccountViewModel.swift]: i don't know   😅

**Review by Mahmoud-Bassuni — APPROVED** [review]

**Review by ZeyadSultan — APPROVED** [review]

**Pull request merged**